src/fields/multi_field.impl.hh

```diff
@@ -45,14 +45,54 @@ template<int spacedim, class Value>
 MultiField<spacedim, Value>::MultiField(const MultiField &other)
 : FieldCommon(other),
   sub_fields_(other.sub_fields_),
-  full_input_list_(other.full_input_list_)
+  full_input_list_(other.full_input_list_),
+  no_check_control_field_(other.no_check_control_field_)
 {
 	this->multifield_ = true;
 }
 
 
 
 template<int spacedim, class Value>
+MultiField<spacedim,Value> &MultiField<spacedim,Value>::operator=(const MultiField<spacedim,Value> &other)
+{
+	ASSERT( flags().match( FieldFlag::input_copy )  , "Try to assign to non-copy field '%s' from the field '%s'.", this->name().c_str(), other.name().c_str());
+	ASSERT(other.shared_->mesh_, "Must call set_mesh before assign to other field.\n");
+	ASSERT( !shared_->mesh_ || (shared_->mesh_==other.shared_->mesh_),
+	        "Assignment between multi fields with different meshes.\n");
+	ASSERT( shared_->comp_names_.size(), "Vector of component names can't be empty!\n");
+	ASSERT( shared_->comp_names_==other.shared_->comp_names_,
+	        "Assignment between multi fields with different vectors of components.\n");
+
+	// check for self assignement
+	if (&other == this) return *this;
+
+	// class members derived from FieldCommon
+	shared_ = other.shared_;
+    shared_->is_fully_initialized_ = false;
+	set_time_result_ = other.set_time_result_;
+	last_time_ = other.last_time_;
+	last_limit_side_ = other.last_limit_side_;
+	is_jump_time_ = other.is_jump_time_;
+	component_index_ = other.component_index_;
+	this->multifield_ = true;
+
+	// class members of Field class
+	if ( size() == other.size() ) {
```

Comment:
What about this:
MultiField<...> a,b;
a.set_components({"x","y"});
b.set_components({"p","q"});
a.setup_components()
b=a; // Thows. But should work.